### PR TITLE
BAU Better data directory handling

### PR DESCRIPTION
**Pull request by j-maynard**
This commit allows us to set set a group on the data directory which will a user to delete their own data directory.  The other advantage of this is it allows us to expire the data directory and replace it.

### CODE REVIEW

File: startup.sh [hunk 1]
```diff
@@ -79,20 +138,37 @@ __     __        _  __         _   _       _        ____  ___
...
+--build-arg GROUP_ID=$(id -g) .
 
 # Inlining the following block of commands to docker run rather than putting
 # them in their own script to avoid an extra level of bash indirection
 docker run -t -v "$script_dir:/verify-local-startup/" verify-local-startup '
 set -e
 if ! test -d data; then
   generate/hub-dev-pki.sh
+  chown -R verify:staff data
+  chown verify:staff *.env
```
Comment (CharlesIC) [inline startup.sh:163]: Why is this needed? What if you run this on a non-GDS machine?
What if the `verify` user doesn't exist?

File: startup.sh [hunk 2]
```diff
@@ -79,20 +138,37 @@ __     __        _  __         _   _       _        ____  ___
...
+
 # Running generate scripts in docker avoids having to install their
 # dependencies on the host.
-docker build -t verify-local-startup .
+docker build -t verify-local-startup \
+--build-arg USER_ID=$(id -u) \
+--build-arg GROUP_ID=$(id -g) .
 
 # Inlining the following block of commands to docker run rather than putting
 # them in their own script to avoid an extra level of bash indirection
```
Comment (CharlesIC) [inline startup.sh:157]: This comment can probably be removed, it doesn't seem to add much

File: startup.sh (hunk 1 above)
Comment (CharlesIC) [reply startup.sh:163]: Isn't it concerning that in order to run a bunch of Java apps the scripts needs to use `chown`? I'm wondering if this scripts is getting too "clever" or is trying to do too many things. It doesn't seem to me that this should be necessary

Comment (j-maynard) [reply startup.sh]: So its actually a work around the issue of Docker... Whenever Docker does anything on Linux with the host operating system it creates files as root because it runs as root.  This means we can't do basics like easily remove the data directory when it gets stale after say a 2 week period.

**Review by instantiator — APPROVED** [review]
